### PR TITLE
Make tracks collapsible and improve capture view rendering.

### PR DESCRIPTION
**Pull request by pierricgimmig**
- Add TriangleToggle class
- Clicking on the white triangle on a track will toggle collapse/expand
- Add tab-like track label with rounded corners
- Removed "m_DesiredHeight" as it was introducing non-uniform scale
- Vertical zoom is temporary disabled until properly reimplemented
- "All tracks" track now drawn after GPU tracks

### CODE REVIEW

**Comment by pierricgimmig** [comment]
CollapsibleTracks

File: OrbitGl/GpuTrack.cpp [hunk 1]
```diff
@@ -20,6 +21,9 @@ GpuTrack::GpuTrack(TimeGraph* time_graph,
   max_time_ = std::numeric_limits<TickType>::min();
 
   string_manager_ = string_manager;
+
+  // Gpu tracks are collapsed by default.
+  collapse_toggle_.SetActive(false);
```
Comment (ronaldfw) [inline OrbitGl/GpuTrack.cpp:26]: This feels a bit like the state is the opposite of what I would it expect to be. I read "collapse toggle active" as "UI element is collapsed", but here it's the inverse.

File: OrbitGl/SchedulerTrack.h [hunk 2]
```diff
@@ -12,6 +12,9 @@ class SchedulerTrack : public ThreadTrack {
   void UpdatePrimitives(uint64_t min_tick, uint64_t max_tick) override;
   Type GetType() const override { return kSchedulerTrack; }
   float GetHeight() const override;
+
+  protected:
+  float GetYFromDepth(float track_y, uint32_t depth, bool collapsed) override;
```
Comment (ronaldfw) [inline OrbitGl/SchedulerTrack.h:17]: While this is not a public API, I would prefer not to use booleans in interfaces as much as possible. Using an enum will make it more obvious what is happening at callsites (and avoids the additional /* collapsed */ comment) and also makes it easier to expand in the future (for example, a track could have states COLLAPSED, SUMMARY, and FULL_DETAIL).

File: OrbitGl/TriangleToggle.h [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+#include <functional>
+
+class TimeGraph;
+class GlCanvas;
+
+class TriangleToggle : public Pickable {
+ public:
+  using StateChangeHandler = std::function<void(bool)>;
+  explicit TriangleToggle(bool initial_state, StateChangeHandler handler,
```
Comment (ronaldfw) [inline OrbitGl/TriangleToggle.h:18]: Another case where having a boolean as a function argument (in this case public API), is unclear. Reading this, I don't know what the two possible initial states mean. Is the toggle collapsed or not? Would also suggest to use an enum here.

File: OrbitGl/Track.cpp [hunk 4]
```diff
@@ -5,73 +5,128 @@
...
-                  a_Canvas->GetPickingManager().CreatePickableId(this))
-            : glColor4ubv(&m_Color[0]);
+std::vector<Vec2> GetRoundedCornerMask(float radius, uint32_t num_sides) {
+  std::vector<Vec2> points;
+  points.emplace_back(0.f, 0.f);
+  points.emplace_back(0.f, radius);
+
+  float increment_radians = 0.5f * M_PI / static_cast<float>(num_sides);
+  for (uint32_t i = 1; i < num_sides; ++i) {
+    float angle = M_PI + static_cast<float>(i) * increment_radians;
```
Comment (ronaldfw) [inline OrbitGl/Track.cpp:35]: This line creates a warning (which is treated as error) for me (double to float conversion).

Comment (ronaldfw) [reply OrbitGl/Track.cpp:35]: (And line 33 as well.)

**Comment by ronaldfw** [comment]
I noticed some bugs when trying this out for myself, probably easier to discuss in person, but in brief: 
- Tracks with dynamically instrumented function events don't collapse for me.
- It's hard to hit the triangle, maybe just make the hitbox larger.
- Some minor layout issues here and there (I'll follow up).

(But overall it's really cool and looks really nice.)

**Comment by ronaldfw** [comment]
> * Tracks with dynamically instrumented function events don't collapse for me.

Sorry, this is wrong, I only had a depth of 1, in which case collapsing doesn't do anything. 

(Maybe we should not show the triangle button when collapsing doesn't do anything?)

File: OrbitGl/CoreMath.h [hunk 5]
```diff
@@ -3,7 +3,10 @@
 //-----------------------------------
 #pragma once
 
+#define _USE_MATH_DEFINES
```
[the remark this answers is not in the document]
Comment (pierricgimmig) [reply OrbitGl/CoreMath.h]: Yeah let's define our own.  I added kPiFloat.

File: OrbitGl/GpuTrack.cpp (hunk 1 above)
Comment (pierricgimmig) [reply OrbitGl/GpuTrack.cpp]: I agree, I changed it to an enum which is much more explicit.

File: OrbitGl/TriangleToggle.h (hunk 3 above)
Comment (pierricgimmig) [reply OrbitGl/TriangleToggle.h]: Done.

File: OrbitGl/Track.cpp (hunk 4 above)
Comment (pierricgimmig) [reply OrbitGl/Track.cpp]: Added kPiFloat constant.

**Comment by pierricgimmig** [comment]
Thanks Ronald for the detailed comments.  I addressed the issues and also added some more roundness on the far right side of the tracks.  Also, the triangle is now a lot easier to toggle, the triangle is greyed out if the tracks can't be collapsed, the GPU tracks are symmetric when collapsed, and various little fixes.

**Comment by ronaldfw** [comment]
Very cool, thanks!

**Pull request merged**